src/app/core/basic-datatypes/configurable-enum/configurable-enum.service.ts

```diff
@@ -47,4 +49,10 @@ export class ConfigurableEnumService {
     }
     return this.enums.get(entityId);
   }
+  private addNewOption(name: string) {
+    const option = { id: name, label: name };
+    this.enumEntity.values.push(option);
+    this.entityMapper.save(this.enumEntity);
+    return option;
+  }
```

Comment:
Its not necessary to add this function here. When the popup is closed, it automatically saves the whole entity enum entity. All that needs to be done is to add the new option to the enum.